Demo/src/main/java/com/paypal/android/api/interceptors/TokenRefreshInterceptor.kt

```diff
@@ -0,0 +1,40 @@
+package com.paypal.android.api.interceptors
+
+import android.util.Log
+import com.paypal.android.api.model.AuthToken
+import com.paypal.android.api.services.AuthApi
+import kotlinx.coroutines.runBlocking
+import okhttp3.Interceptor
+import okhttp3.Request
+import okhttp3.Response
+import javax.inject.Inject
+
+class TokenRefreshInterceptor @Inject constructor(
+    private val authApi: AuthApi,
+) : Interceptor {
+
+    companion object {
+        private const val EXPIRATION_FACTOR = 0.95
+    }
+
+    private val tag = this::class.java.toString()
+
+    override fun intercept(chain: Interceptor.Chain): Response {
+        val mainRequest = chain.request()
+        var authToken = AuthApi.authToken
+        val now = System.currentTimeMillis()
+        if (authToken == null || authToken.expiresAt < now) {
+            Log.i(tag, "Creating a new OAuth Token...")
+            val token = runBlocking { authApi.postOAuthToken() } // refreshToken()
```

Comment:
I think we should wrap this `runBlocking` call in a try/catch block since it may throw an exception if the service call fails

Reply:
@sshropshire @ndmunoz10 this where I overreached lol
I implemented this when the sample server wasnt ready yet. Now that it is, we dont need any of this. BUT, it will come in handy once we implement `Client side` integrations. Right now this code is not being called, but i kept it there because we will need it.

Reply:
Haha all good. It is technically versioned though if we need it again we can reach back into git. I think it makes sense to keep only the code we need in `main` since it's an additional maintenance cost even if it isn't being used.

Reply:
@sshropshire i'll branch it out and keep it there, im guessing we'll need it soon